spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -48,15 +46,9 @@
 
 	private AckMode ackMode = AckMode.AUTO;
 
-	private MessageConverter messageConverter;
```

Comment:
That's not good change.
The `PubSubInboundChannelAdapter` hasn't been designed to only support Binder.

Many people use Channel Adapters as is, without any Spring Cloud Stream magic.
So, such a convert was a good functionality in this component and you just remove it without any explanation.

If you would like support `byte[]` as a `payload` to emit, would be better to improve the logic in this component like `PathThroughConverter` or even better to let to support `null` for the converter. 

But I'm against such a critical change that narrows a component functionality for nothing.

Thanks for understanding.

(Again no any commit comments on the matter)

Reply:
Indeed, the motivation for this was that the previous converter couldn't convert `application/json` messages coming from the binder.
I tried to also add a `MappingJackson2MessageConverter` into a `CompositeMessageConverter`, but apparently SCSt re-applies internal message conversion after `sendMessage()`, which it couldn't do after the message had been re-converted by our converter here.

That's when I saw it made sense to just pass through the `byte[]` that we receive and just send it as-is to the output channel.

The user could then parse the `byte[]` on their end to whatever they want. I agree this is far from ideal, but the current functionality is also not very far from this... We basically consider every payload to be a string, which is not right.

I couldn't find anything on `PathThroughConverter`.

Reply:
That's why I suggest and idea to let to path the `converter` into the `PubSubInboundChannelAdapter` as `null` to allow to return the body as a `byte[]`. 
This is much flexible and functional.
The end-user then has a choice to decide what the conversion model to select or jut deal with `byte[]`. The Binder then just will passes the `null` to the converter property.

That's exactly the way I've chosen for the Kinesis Binder: https://github.com/spring-cloud/spring-cloud-stream-binder-aws-kinesis/blob/master/spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/KinesisMessageChannelBinder.java#L203

A `PathThroughConverter` is just an idea you can implement, however the `null` is much performant 😄 

Nevertheless you didn't explain the change in the commit message...

Reply:
Ideally, I would like to have a `CompositeMessageConverter` here similarly to SCSt that contains a bunch of pre-configured `MessageConverters` that convert payloads according to their content types.
That way we could support a multitude of payloads. Although for SCSt, this could possibly be overkill or even now work.

What do you think of having `messageConverter` always null, and if the users are interested in any specific kind of conversion/payload, they can specify the converter on demand?

Reply:
Well, for consistency with other Channel Adapters in the Spring Integration I would insist on the default convenient converter, but allow to configure it to `null` to let to deal with the raw `byte[]` as it is in case of SCSt. 

We can have any possible converter complexity, but that's really not productive to convert on the adapter and then on the SCSt side.